generator/internal/connection_impl_generator.cc

```diff
@@ -119,39 +119,25 @@ class $connection_class_name$Impl
   // So, we have to fallback to `options_`.
   HeaderPrint(R"""(
  private:
-  std::unique_ptr<$product_namespace$::$retry_policy_name$> retry_policy() {
-    auto const& options = internal::CurrentOptions();
-    if (options.has<$product_namespace$::$retry_policy_name$Option>()) {
-      return options.get<$product_namespace$::$retry_policy_name$Option>()->clone();
-    }
-    return options_.get<$product_namespace$::$retry_policy_name$Option>()->clone();
+  static std::unique_ptr<$product_namespace$::$retry_policy_name$>
+  retry_policy(Options const& options) {
+    return options.get<$product_namespace$::$retry_policy_name$Option>()->clone();
   }
```

Comment:
I see two checks, one on the thread-local options and one on the member options, being replaced by a single check, just on the thread-local options.  But I don't think I see anything (at least here) that compensates for that reduction.

Reply:
I guess that just echoes the "We may need to update some unit tests" statement in #9168.

Reply:
> But I don't think I see anything (at least here) that compensates for that reduction.

Other than in unit tests, the thread-local options is a superset of `options_`. As you probably noted, I changed the unit tests when needed.